test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/StackExchangeRedisCallsInstrumentationTests.cs

```diff
@@ -203,13 +204,44 @@ public void StackExchangeRedis_BadArgs()
             Assert.Throws<ArgumentNullException>(() => builder.AddRedisInstrumentation(null));
 
             var activityProcessor = new Mock<BaseProcessor<Activity>>();
-            Assert.Throws<ArgumentNullException>(() =>
+            Assert.Throws<NotSupportedException>(() =>
             Sdk.CreateTracerProviderBuilder()
                 .AddProcessor(activityProcessor.Object)
                 .AddRedisInstrumentation(null)
                 .Build());
         }
 
+        [Fact]
+        public void StackExchangeRedis_DependencyInjection_Success()
+        {
+            var connectionOptions = new ConfigurationOptions
+            {
+                AbortOnConnectFail = false,
+            };
+            connectionOptions.EndPoints.Add("localhost");
+
+            var services = new ServiceCollection();
+            services.AddSingleton<IConnectionMultiplexer>(ConnectionMultiplexer.Connect(connectionOptions));
+
+            services.AddOpenTelemetryTracing(builder => builder.AddRedisInstrumentation());
+
+            using var serviceProvider = services.BuildServiceProvider();
+
+            var tracerProvider = serviceProvider.GetRequiredService<TracerProvider>();
```

Comment:
Do we want to test that the `DeferredTracerProviderBuilder` is correctly adding `RedisInstrumentation` by assigning some variable in `configure` action and assert that fact. Something like this: https://github.com/open-telemetry/opentelemetry-dotnet/blob/2347c48203492d05716d580f8310c94057d3195c/test/OpenTelemetry.Instrumentation.AspNetCore.Tests/DependencyInjectionConfigTests.cs#L70

Reply:
Updated